pytorch_lightning/utilities/cli.py

```diff
@@ -166,7 +166,8 @@ def __init__(
     def setup(self, trainer: Trainer, pl_module: LightningModule, stage: Optional[str] = None) -> None:
         # save the config in `setup` because (1) we want it to save regardless of the trainer function run
         # and we want to save before processes are spawned
-        log_dir = trainer.log_dir or trainer.default_root_dir
+        log_dir = trainer.log_dir
+        assert log_dir is not None
```

Comment:
IMO, I don't like assert in code without explication. Please, raise an exception

Reply:
@tchaton This is to make mypy happy, not because we expect `log_dir` to be `None`